src/CacheManager.Memcached/MemcachedCacheHandle.cs

```diff
@@ -215,6 +215,14 @@ public override bool Exists(string key, string region)
         }
 
         /// <inheritdoc />
+        /// <exception cref="System.NotImplementedException">Not supported in this version.</exception>
+        public override IEnumerable<string> Keys(string pattern, string region)
+        {
+            throw new NotImplementedException();
```

Comment:
I think, always storing all keys in another hashset can be a huge performance issue. Just checked the features of this MemoryCache and it seems Microsoft does not plan to expose the keys collection or an IEnumerable... 
In addition, it seems that keys never get removed from the fake or any other region, meaning, you'll always get all keys which have been deleted at some point, which is bad.
I'm also concerned about how that behaves in scenarios where multiple threads manipulate that hashset...

I'd say, either add a configuration option to explicitly enable the ` FindKeys` feature, or don't implement it at all for this cache